src/sources/dnstap/parser.rs

```diff
@@ -986,43 +988,92 @@ mod tests {
             .expect("Invalid base64 encoded data.");
         let parse_result = DnstapParser::parse(&mut log_event, Bytes::from(dnstap_data));
         assert!(parse_result.is_ok());
-        assert!(log_event
-            .all_fields()
-            .unwrap()
-            .any(|(key, value)| key == "time"
-                && match *value {
-                    Value::Integer(time) => time == 1_593_489_007_920_014_129,
-                    _ => false,
-                }));
-        assert!(log_event
-            .all_fields()
-            .unwrap()
-            .any(|(key, value)| key == "timestamp"
-                && match *value {
-                    Value::Timestamp(timestamp) =>
-                        timestamp.timestamp_nanos() == 1_593_489_007_920_014_129,
-                    _ => false,
-                }));
-        assert!(log_event
-            .all_fields()
-            .unwrap()
-            .any(|(key, value)| key == "requestData.header.qr"
-                && match *value {
-                    Value::Integer(qr) => qr == 0,
-                    _ => false,
-                }));
-        assert!(log_event.all_fields().unwrap().any(|(key, value)| key
-            == "requestData.opt.udpPayloadSize"
-            && match *value {
-                Value::Integer(udp_payload_size) => udp_payload_size == 512,
-                _ => false,
-            }));
-        assert!(log_event.all_fields().unwrap().any(|(key, value)| key
-            == "requestData.question[0].domainName"
-            && match value {
-                Value::Bytes(domain_name) => *domain_name == Bytes::from_static(b"facebook1.com."),
-                _ => false,
-            }));
+
+        let expected_map: BTreeMap<&str, Value> = BTreeMap::from([
```

Comment:
@StephenWakely to increase confidence that this PR didn't break anything I added these extra checks.